website/content/en/docs/olm-integration/legacy/generating-a-csv.md

```diff
@@ -0,0 +1,208 @@
+
+By default `generate bundle` will generate a CSV, copy CRDs, and generate metadata in the bundle format:
+
+```console
+$ operator-sdk generate bundle --version 0.0.1
+$ tree ./deploy/olm-catalog/test-operator
+./deploy/olm-catalog/test-operator
+├── manifests
+│   ├── cache.my.domain_memcacheds.yaml
+│   └── memcached-operator.clusterserviceversion.yaml
+└── metadata
+    └── annotations.yaml
+```
+
+### Package manifests format
+
+A [package manifests][package-manifests] format consists of on-disk manifests (CSV and CRDs) and metadata that
+define an Operator at all versions of that Operator. Each version is contained in its own directory, with a parent
+package manifest YAML file containing channel-to-version mappings, much like a bundle's metadata.
+
+By default `generate packagemanifests` will generate a CSV, a package manifest file, and copy CRDs in the
+[package manifests][package-manifests] format:
+
+```console
+$ operator-sdk generate bundle --version 0.0.1
+$ tree ./deploy/olm-catalog/test-operator
+./deploy/olm-catalog/test-operator
+├── 0.0.1
+│   ├── cache.my.domain_memcacheds.yaml
+│   └── memcached-operator.clusterserviceversion.yaml
+└── memcached-operator.package.yaml
+```
+
+## Update your Operator
+
+Lets say you added a new API `App` with group `app.example.com` and version `v1alpha1` to your Operator project,
+and added a port to your manager Deployment in `deploy/operator.yaml`.
+
+If using a bundle format, the current version of your CSV can be updated by running:
+
+```console
+$ operator-sdk generate bundle
+```
+
+If using a package manifests format, run:
+
+```console
+$ operator-sdk generate packagemanifests --version 0.0.1
+```
+
+Running the command for either format will append your new CRD to `spec.customresourcedefinitions.owned`,
+replace the old data at `spec.install.spec.deployments` with your updated Deployment,
+and update your existing CSV manifest. The SDK will not overwrite [user-defined](#csv-fields)
+fields like `spec.maintainers`.
+
+## Upgrade your Operator
+
+Lets say you're upgrading your Operator to version `v0.0.2`. You also want to add a new channel `beta`,
+and use it as the default channel.
+
+If using a bundle format, a new version of your CSV can be created by running:
+
+```console
+$ operator-sdk generate bundle --version 0.0.2 --channels=beta --default-channel=beta
+```
+
+If using a package manifests format, run:
+
+```console
+$ operator-sdk generate packagemanifests --from-version 0.0.1 --version 0.0.2 --channel=beta --default-channel
+```
+
+Running the command for either format will persist user-defined fields, updates `spec.version`,
+and populates `spec.replaces` with the old CSV versions' name.
+
+## CSV fields
+
+Below are two lists of fields: the first is a list of all fields the SDK and OLM expect in a CSV, and the second are optional.
+
+_For Go Operators only:_ Several fields require user input (labeled _user_) or a [CSV marker][csv-markers] (labeled _marker_).
```

Comment:
Should we also make it clear that Helm and Ansible operators do not support marker-generated fields?